Pass along authorization token for touch events

pascoej opened the pull request:
#### 312e1f95f86f42d48b464e54109375a036572cde
<pre>
Pass along authorization token for touch events
<a href="https://bugs.webkit.org/show_bug.cgi?id=286775">https://bugs.webkit.org/show_bug.cgi?id=286775</a>
<a href="https://rdar.apple.com/problem/143917525">rdar://problem/143917525</a>

Reviewed by Charlie Wolfe.

We pass along authorization tokens for mouse and keyboard events for later verifying in the UI process, we should do the same for touch events.

I have added a layout test to exercise this behavior. This also requires an internal change that is related in the radar.

* LayoutTests/fast/events/ios/single-tap-verify-user-gesture-expected.txt: Added.
* LayoutTests/fast/events/ios/single-tap-verify-user-gesture.html: Added.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleTouchEvent):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
* Source/WebKit/UIProcess/API/APINavigationAction.h:
* Source/WebKit/UIProcess/API/C/WKNavigationActionRef.cpp:
(WKNavigationActionHasUnconsumedUserGesture):
* Source/WebKit/UIProcess/API/C/WKNavigationActionRef.h:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesGetVerifyUserGestureInUIProcessEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendPreventableTouchEvent):
(WebKit::WebPageProxy::sendUnpreventableTouchEvent):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createOtherPage):
(WTR::TestController::createOtherPlatformWebView):

Canonical link: <a href="https://commits.webkit.org/289612@main">https://commits.webkit.org/289612@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04f2c731c0605dca663a0907d115b7ac0559228e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87548 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7063 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41926 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92411 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/38292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89599 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7444 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15231 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/67619 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/38292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90550 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/5688 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79229 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/47973 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/5474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33622 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37405 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/75881 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34487 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94298 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14716 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/10797 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/76456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14970 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75078 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/75678 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18604 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/20058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18483 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/7667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14732 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/20032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14476 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17920 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16258 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on LayoutTests/fast/events/ios/single-tap-verify-user-gesture.html:
```diff
@@ -0,0 +1,52 @@
+    });
+
+    targetForFakeClick.addEventListener("click", (event) => {
+        win = window.open("about:blank");
```
Doesn't this work without your change?

charliewolfe commented on LayoutTests/fast/events/ios/single-tap-verify-user-gesture.html:
```diff
@@ -0,0 +1,52 @@
+(async function() {
+    const targetForFakeClick = document.getElementById("target-fake");
+
+    window.addEventListener("touchend", (event) => {
```
Should make sure we match which events have user gestures in the WP

charliewolfe commented on LayoutTests/fast/events/ios/single-tap-verify-user-gesture.html:
```diff
@@ -0,0 +1,63 @@
+    window.addEventListener("touchend", (event) => {
+        win = window.open("about:blank");
+        if (win && target.innerText == "PASS")
+            target.innerText = "PASS";
```
Wouldn't this override the FAIL text if it was set in the `touchmove` listener?

pascoej replied:
No, b/c this part of the if `target.innerText == "PASS"`

charliewolfe replied:
should change this to “FAIL” when innerText != “PASS” then

pascoej replied:
Nevermind, I see the issue, will update.

charliewolfe commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -530,7 +533,7 @@ void TestController::finishFullscreenExit(WKPageRef page)
 }
 
-PlatformWebView* TestController::createOtherPlatformWebView(PlatformWebView* parentView, WKPageConfigurationRef configuration, WKNavigationActionRef, WKWindowFeaturesRef)
+PlatformWebView* TestController::createOtherPlatformWebView(PlatformWebView* parentView, WKPageConfigurationRef configuration, WKNavigationActionRef navigationAction, WKWindowFeaturesRef)
```
undo

pascoej committed:
Pass along authorization token for touch events
https://bugs.webkit.org/show_bug.cgi?id=286775
rdar://problem/143917525

Reviewed by Charlie Wolfe.

We pass along authorization tokens for mouse and keyboard events for later verifying in the UI process, we should do the same for touch events.

I have added a layout test to exercise this behavior. This also requires an internal change that is related in the radar.

* LayoutTests/fast/events/ios/single-tap-verify-user-gesture-expected.txt: Added.
* LayoutTests/fast/events/ios/single-tap-verify-user-gesture.html: Added.
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::handleTouchEvent):
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::WebKit2PlatformTouchEvent::WebKit2PlatformTouchEvent):
* Source/WebKit/UIProcess/API/APINavigationAction.h:
* Source/WebKit/UIProcess/API/C/WKNavigationActionRef.cpp:
(WKNavigationActionHasUnconsumedUserGesture):
* Source/WebKit/UIProcess/API/C/WKNavigationActionRef.h:
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesGetVerifyUserGestureInUIProcessEnabled):
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::sendPreventableTouchEvent):
(WebKit::WebPageProxy::sendUnpreventableTouchEvent):
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::createOtherPage):
(WTR::TestController::createOtherPlatformWebView):

Canonical link: https://commits.webkit.org/289612@main
